alter streaming jobs with dependencies (#1780)
risingwavelabs/risingwave-docs: docs/manage/alter-streaming.md

    
```sql
CREATE MATERIALIZED VIEW cust_sales AS
    SELECT
        customer_id,
        SUM(total_price) AS sales_amount
    FROM orders
    GROUP BY customer_id;
```

Here we create a new materialized view `cust_sales_new` with the new column `sales_count`:
    
```sql
CREATE MATERIALIZED VIEW cust_sales_new AS
    SELECT
        customer_id,
        SUM(total_price) AS sales_amount,
        COUNT(*) AS sales_count -- The new column
    FROM orders
    GROUP BY customer_id;
```

After the new materialized view is created, we can drop the old materialized view `cust_sales` and rename `cust_sales_new` to `cust_sales`:

```sql
DROP MATERIALIZED VIEW cust_sales;
ALTER MATERIALIZED VIEW cust_sales_new RENAME TO cust_sales;
```

## Alter a sink

To alter a sink, you will need to create a new sink and drop the old sink. Please check the example from the last section.

If your sink is created based on a materialized view using the `CREATE SINK ... FROM ...` statement, you have the option to specify `without_backfill = true` to exclude existing data.

```sql
CREATE SINK ... FROM ... WITH (without_backfill = true).
```

## Alter streaming jobs with dependencies

> If the modified materialized view is depended upon by other materialized views or sinks, or if multiple materialized views or sinks with dependencies need to be modified simultaneously, similarly, first create all the new ones and then drop all the old ones.

Reviewer: @WanYixian Please help reword this paragraph. You can see the example below for reference. If A is the level-0 MV, and B is level-1 MV, meaning B is built on top of A, and C is level-2 MV. To modify A, we need to create A_new, then B_new based on A_new, and C_new based on B_new, and then drop A, B, and C, and then rename A_new to A, B_new to B, and C_new to C.

    
```sql
CREATE MATERIALIZED VIEW cust_sales AS
    SELECT
        customer_id,
        SUM(total_price) AS sales_amount
    FROM orders
    GROUP BY customer_id;
```

Here we create a new materialized view `cust_sales_new` with the new column `sales_count`:
    
```sql
CREATE MATERIALIZED VIEW cust_sales_new AS
    SELECT
        customer_id,
        SUM(total_price) AS sales_amount,
        COUNT(*) AS sales_count -- The new column
    FROM orders
    GROUP BY customer_id;
```

After the new materialized view is created, we can drop the old materialized view `cust_sales` and rename `cust_sales_new` to `cust_sales`:

```sql
DROP MATERIALIZED VIEW cust_sales;
ALTER MATERIALIZED VIEW cust_sales_new RENAME TO cust_sales;
```

## Alter a sink

To alter a sink, you will need to create a new sink and drop the old sink. Please check the example from the last section.

If your sink is created based on a materialized view using the `CREATE SINK ... FROM ...` statement, you have the option to specify `without_backfill = true` to exclude existing data.

```sql
CREATE SINK ... FROM ... WITH (without_backfill = true).
```

## Alter streaming jobs with dependencies

> If the modified materialized view is depended upon by other materialized views or sinks, or if multiple materialized views or sinks with dependencies need to be modified simultaneously, similarly, first create all the new ones and then drop all the old ones.

Reviewer 2: ```suggestion
If other materialized views or sinks depend upon the modified materialized view, or if multiple materialized views or sinks with dependencies need updating simultaneously, similarly, create all the new ones and then drop all the old ones.
```
